core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -297,6 +297,10 @@ public static boolean deleteFile(String filePath) throws IOException {
     return getCarbonFile(filePath).deleteFile();
   }
 
+  public static boolean deleteFile(CarbonFile carbonFile) throws IOException {
```

Comment:
Yes, it is there but it tries to get the carbonFile first, in case there is carbonFile already present in the parent method, it will again get it, which can be avoided and is more optimised. There is even a JIRA opened by Ajantha to do the exact same thing, so that we reduce unnecessary calls to getCarbonFIle.

Reply:
Please remove `deleteFile(CarbonFile carbonFile)`, because there is no extra function is happening here. directly call carbonFile.deleteFile() in the usages.

If there was some extra functionality, then it makes sense to have to APIfor carbonFile also to avoid getting CarbonFile again. for this case it is not needed

Reply:
removed